oslo_aero_3_0a007: Resolve pcols / ncol conflict

gold2718 opened the pull request:
Fix mismatch where some arrays in an equation used `:` as the first dimension while others used `:ncol`. This fails when `ncol` /= `pcols` which happens when there is an unfilled chunk on any task.

Addresses NorESMhub/CAM#233

Commit:
resolve pcols / ncol conflict

gold2718 commented:
@TomasTorsvik, can you try this branch in place of the current tag? In the meantime, I need to modify at least one aux_cam_noresm test to have an `ncol` / `pcols` mismatch so problems like this can be found earlier. This new test can then be part of the new CAM tag that incorporates this fix.

Commit:
Fix outfld call, was sending invalid values (pcols => ncol)
Limit output to atm.log

TomasTorsvik commented:
@gold2718 - the two tests that were failing in https://github.com/NorESMhub/CAM/issues/233 are passing when I include this patc for OSLO_AERO.

Commit:
Fix dimensions of ccn

ccn had pverp levels instead of pver
This may fix https://github.com/NorESMhub/CAM/issues/180

gold2718 commented:
@oyvindseland, this bug fix PR might also fix NorESMhub/CAM#180

gold2718 commented:
> @gold2718 - the two tests that were failing in [NorESMhub/CAM#233](https://github.com/NorESMhub/CAM/issues/233) are passing when I include this patc for OSLO_AERO.

That's great. It turns out that this bug is not caught by the Intel compiler. When I switched to gnu, I was able to reproduce this problem in the aux_cam_noresm tests as well as another error that I fixed (and which might fix NorESMhub/CAM#180).

Given all that, do you feel like you could approve this PR so I can make a new tag and a new CAM tag that uses the fix?
